reference/constraints/Week.rst

```diff
@@ -0,0 +1,172 @@
+
+Validates that a string (or an object implementing the ``Stringable`` PHP interface)
+matches a given week number. The week number format is defined by `ISO-8601`_
+and should be composed of the year and the week number, separated by a hyphen
+(e.g. ``2022-W01``).
+
+==========  =======================================================================
+Applies to  :ref:`property or method <validation-property-target>`
+Class       :class:`Symfony\\Component\\Validator\\Constraints\\Week`
+Validator   :class:`Symfony\\Component\\Validator\\Constraints\\WeekValidator`
+==========  =======================================================================
+
+Basic Usage
+-----------
+
+If you wanted to ensure that the ``startWeek`` property of an ``OnlineCourse``
+class is between the first and the twentieth week of the year 2022, you could do
+the following:
+
+.. configuration-block::
+
+    .. code-block:: php-attributes
+
+        // src/Entity/OnlineCourse.php
+        namespace App\Entity;
+
+        use Symfony\Component\Validator\Constraints as Assert;
+
+        class OnlineCourse
+        {
+            #[Assert\Week(min: '2022-W01', max: '2022-W20')]
+            protected string $startWeek;
+        }
+
+    .. code-block:: yaml
+
+        # config/validator/validation.yaml
+        App\Entity\OnlineCourse:
+            properties:
+                startWeek:
+                    - Week:
+                        min: '2022-W01'
+                        max: '2022-W20'
+
+    .. code-block:: xml
+
+        <!-- config/validator/validation.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <constraint-mapping xmlns="http://symfony.com/schema/dic/constraint-mapping"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/constraint-mapping https://symfony.com/schema/dic/constraint-mapping/constraint-mapping-1.0.xsd">
+
+            <class name="App\Entity\OnlineCourse">
+                <property name="startWeek">
+                    <constraint name="Week">
+                        <option name="min">2022-W01</option>
+                        <option name="max">2022-W20</option>
+                    </constraint>
+                </property>
+            </class>
+        </constraint-mapping>
+
+    .. code-block:: php
+
+        // src/Entity/OnlineCourse.php
+        namespace App\Entity;
+
+        use Symfony\Component\Validator\Constraints as Assert;
+        use Symfony\Component\Validator\Mapping\ClassMetadata;
+
+        class OnlineCourse
+        {
+            // ...
+
+            public static function loadValidatorMetadata(ClassMetadata $metadata): void
+            {
+                $metadata->addPropertyConstraint('startWeek', new Assert\Week([
+                    'min' => '2022-W01',
+                    'max' => '2022-W20',
+                ]));
+            }
+        }
+
+.. tip::
```

Comment:
```suggestion
.. note::
```

Reply:
Do we use tip anywhere else?

Reply:
It can happen, but you're right, a note is more appropriate here